src/Hpack/Config.hs

```diff
@@ -416,12 +442,12 @@ data Library = Library {
 } deriving (Eq, Show)
 
 data Executable = Executable {
-  executableName :: String
-, executableMain :: FilePath
+  executableName :: Maybe String
```

Comment:
`executableName` can never be `Nothing` and is never valid inside a conditional.  So what we want here is a type that represents this properly.  There are probably several ways to achieve this, what I did is:

Change `[Section Executable]` to `Map String (Section Executable)` in `Package` and remove `executableName` from `Executable`.  So what we have now is a mapping from executable names to executable sections.